holoviews/tests/operation/test_datashader.py

```diff
@@ -1371,26 +1372,73 @@ def test_rasterize_summerize(point_plot):
 
     # Count is from the aggregator
     assert list(img.data) == ["Count", "__index__", "s", "val", "cat"]
-    assert list(img.vdims) == ["Count", "s", "val", "cat"]  # no index
+    assert list(img.vdims) == [Dimension("Count")]  # Only the dimension send to the frontend
 
     # The output for the selector should be equal to the output for the aggregator using
     # ds.where
-    img_agg = rasterize(point_plot, aggregator=ds.where(sel_fn("val")), **rast_input)
+    img_agg = rasterize(point_plot, aggregator=ds.where(sel_fn("val")), **inputs)
     for c in ["s", "val", "cat"]:
-        np.testing.assert_array_equal(img[c], img_agg[c])
+        np.testing.assert_array_equal(img.data[c], img_agg.data[c], err_msg=c)
 
     # Checking the count is also the same
-    img_count = rasterize(point_plot, **rast_input)
+    img_count = rasterize(point_plot, **inputs)
     np.testing.assert_array_equal(img["Count"], img_count["Count"])
 
+@pytest.mark.parametrize("sel_fn", (ds.first, ds.last, ds.min, ds.max))
+def test_selector_datashade(point_plot, sel_fn):
+    inputs = dict(dynamic=False,  x_range=(-1, 1), y_range=(-1, 1), width=10, height=10)
+    img = datashade(point_plot, selector=sel_fn("val"), **inputs)
+
+    # RGBA is from the aggregator
+    assert list(img.data) == [*"RGBA", "__index__", "s", "val", "cat"]
+    assert list(img.vdims) == [*map(Dimension, "RGBA")]  # Only the RGBA send to the frontend
+
+    # The output for the selector should be equal to the output for the aggregator using
+    # ds.where
+    img_agg = rasterize(point_plot, aggregator=ds.where(sel_fn("val")), **inputs)
+    for c in ["s", "val", "cat"]:
+        np.testing.assert_array_equal(img.data[c], img_agg.data[c], err_msg=c)
+
+    # Checking the RGBA is also the same
+    img_count = datashade(point_plot, **inputs)
+    for n in "RGBA":
+        np.testing.assert_array_equal(img[n], img_count[n], err_msg=n)
 
-def test_rasterize_with_datetime_column():
+
+@pytest.mark.parametrize("op_fn", (rasterize, datashade))
+@pytest.mark.parametrize(
+    "agg_fn", (ds.count(), ds.by("cat")), ids=["count", "by"]
+)
+def test_selector_spread(point_plot, op_fn, agg_fn):
+    inputs = dict(dynamic=False,  x_range=(-1, 1), y_range=(-1, 1), width=10, height=10)
+    img = op_fn(point_plot, aggregator=agg_fn, selector=ds.first("val"), **inputs)
+    spread_img = spread(img)
+
+    with suppress(AssertionError): # We expect them to be different
+        xr.testing.assert_equal(spread_img.data, img.data)
+        raise ValueError("The spread should not be equal to the original image")
+
+    with suppress(AssertionError): # We expect them to be different
+        np.testing.assert_array_equal(spread_img.data["__index__"], img.data["__index__"])
+        raise ValueError("The spread should not be equal to the original image")
+
+    if isinstance(agg_fn, ds.count):
+        data = spread_img.vdims[-1].name  # FIXME: Last one as it is alpha for datashade, shouldn't matter
```

Comment:
This is the last thing that I need to look into. RGBA gives non-zero when alpha is zero:

![image](https://github.com/user-attachments/assets/786ace68-1b01-41d1-8216-3fc167e8fc94)

Reply:
I have resolved this in 9e719101d. I think it is correct to set it to 0 when alpha is zero, but it is likely to be a somewhat subjective opinion.

Reply:
Doesn't much matter in the end, I think we should hide the RGBA value on hover in most cases now that we have real hover info.